src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -0,0 +1,591 @@
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+#if UNIFIED
+using CoreFoundation;
+using Foundation;
+using Security;
+#else
+using MonoTouch.CoreFoundation;
+using MonoTouch.Foundation;
+using MonoTouch.Security;
+using System.Globalization;
+#endif
+
+#if SYSTEM_NET_HTTP
+namespace System.Net.Http {
+#else
+namespace Foundation {
+#endif
+	public class NSUrlSessionHandler : HttpMessageHandler
+	{
+		private readonly Dictionary<string, string> headerSeparators = new Dictionary<string, string> {
+			["User-Agent"] = " ",
+			["Server"] = " "
+		};
+
+		private readonly NSUrlSession session;
+		private readonly Dictionary<NSUrlSessionTask, InflightData> inflightRequests;
+		private readonly object inflightRequestsLock = new object ();
+
+		public NSUrlSessionHandler ()
+		{
+			var configuration = NSUrlSessionConfiguration.DefaultSessionConfiguration;
+
+			// we cannot do a bitmask but we can set the minimum based on ServicePointManager.SecurityProtocol minimum
+			var sp = ServicePointManager.SecurityProtocol;
+			if ((sp & SecurityProtocolType.Ssl3) != 0)
+				configuration.TLSMinimumSupportedProtocol = SslProtocol.Ssl_3_0;
+			else if ((sp & SecurityProtocolType.Tls) != 0)
+				configuration.TLSMinimumSupportedProtocol = SslProtocol.Tls_1_0;
+			else if ((sp & SecurityProtocolType.Tls11) != 0)
+				configuration.TLSMinimumSupportedProtocol = SslProtocol.Tls_1_1;
+			else if ((sp & SecurityProtocolType.Tls12) != 0)
+				configuration.TLSMinimumSupportedProtocol = SslProtocol.Tls_1_2;
+
+			session = NSUrlSession.FromConfiguration (NSUrlSessionConfiguration.DefaultSessionConfiguration, new NSUrlSessionHandlerDelegate (this), null);
+			inflightRequests = new Dictionary<NSUrlSessionTask, InflightData> ();
+		}
+
+		protected override void Dispose (bool disposing)
```

Comment:
This looks redundant, since it's just calling the base method. Can't this override just be removed?